### ISSUE TITLE
Crash when forcing unlock

### ISSUE THREAD

**Issue by gusfune** [opentofu/opentofu#1617]
### OpenTofu Version

```shell
OpenTofu v1.7.0
on darwin_arm64
+ provider registry.opentofu.org/cloudflare/cloudflare v4.30.0
+ provider registry.opentofu.org/dopplerhq/doppler v1.7.0
+ provider registry.opentofu.org/hashicorp/archive v2.4.2
+ provider registry.opentofu.org/hashicorp/aws v5.46.0
+ provider registry.opentofu.org/hashicorp/google v5.25.0
+ provider registry.opentofu.org/hashicorp/random v3.6.1
+ provider registry.opentofu.org/vercel/vercel v1.6.0
```

### OpenTofu Configuration Files

```hcl
terraform {
  backend "gcs" {
    bucket = "amazing-bucket"
    prefix = "terraform/state"
  }
}

terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 5.0"
    }

    doppler = {
      source  = "DopplerHQ/doppler"
      version = "~> 1.7"
    }

    google = {
      source  = "hashicorp/google"
      version = "~> 5"
    }

    cloudflare = {
      source  = "cloudflare/cloudflare"
      version = "~> 4"
    }

    vercel = {
      source  = "vercel/vercel"
      version = "~> 1"
    }
  }
}

provider "aws" {
  region = "us-east-1"
}

provider "cloudflare" {
  api_token = var.cloudflare_api_token
}

provider "doppler" {
  doppler_token = var.doppler_token
}

provider "google" {
  project = "amazing-project"
}

provider "vercel" {
  api_token = var.vercel_token
  team      = "foo-bar"
}

```

rest is irrelevant for replication due to the problem being a state lock

### Debug Output

```
2024-05-07T09:14:25.409+0100 [INFO]  OpenTofu version: 1.7.0
2024-05-07T09:14:25.409+0100 [DEBUG] using github.com/hashicorp/go-tfe v1.36.0
2024-05-07T09:14:25.409+0100 [DEBUG] using github.com/opentofu/hcl/v2 v2.0.0-20240416130056-03228b26f391
2024-05-07T09:14:25.409+0100 [DEBUG] using github.com/hashicorp/terraform-svchost v0.1.1
2024-05-07T09:14:25.409+0100 [DEBUG] using github.com/zclconf/go-cty v1.14.4
2024-05-07T09:14:25.409+0100 [INFO]  Go runtime version: go1.22.2
2024-05-07T09:14:25.409+0100 [INFO]  CLI args: []string{"tofu", "force-unlock", "1714841031528627"}
2024-05-07T09:14:25.409+0100 [TRACE] Stdout is a terminal of width 173
2024-05-07T09:14:25.409+0100 [TRACE] Stderr is a terminal of width 173
2024-05-07T09:14:25.409+0100 [TRACE] Stdin is a terminal
2024-05-07T09:14:25.409+0100 [DEBUG] Attempting to open CLI config file: /Users/gus/.tofurc
2024-05-07T09:14:25.409+0100 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2024-05-07T09:14:25.410+0100 [INFO]  Loading CLI configuration from /Users/gus/.terraform.d/credentials.tfrc.json
2024-05-07T09:14:25.410+0100 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2024-05-07T09:14:25.410+0100 [DEBUG] ignoring non-existing provider search directory /Users/gus/.terraform.d/plugins
2024-05-07T09:14:25.410+0100 [DEBUG] ignoring non-existing provider search directory /Users/gus/Library/Application Support/io.terraform/plugins
2024-05-07T09:14:25.410+0100 [DEBUG] ignoring non-existing provider search directory /Library/Application Support/io.terraform/plugins
2024-05-07T09:14:25.411+0100 [DEBUG] Found the config directory: /Users/gus/.terraform.d
2024-05-07T09:14:25.411+0100 [INFO]  CLI command args: []string{"force-unlock", "1714841031528627"}
2024-05-07T09:14:25.425+0100 [TRACE] Meta.Backend: built configuration for "gcs" backend with hash value 3845035542
2024-05-07T09:14:25.425+0100 [TRACE] Preserving existing state lineage "368a9b13-7393-a69f-7f42-493fe3fa04f9"
2024-05-07T09:14:25.425+0100 [TRACE] Preserving existing state lineage "368a9b13-7393-a69f-7f42-493fe3fa04f9"
2024-05-07T09:14:25.426+0100 [TRACE] Meta.Backend: working directory was previously initialized for "gcs" backend
2024-05-07T09:14:25.426+0100 [TRACE] Meta.Backend: using already-initialized, unchanged "gcs" backend configuration
2024-05-07T09:14:25.428+0100 [TRACE] Meta.Backend: instantiated backend of type *gcs.Backend
2024-05-07T09:14:25.430+0100 [TRACE] providercache.fillMetaCache: scanning directory .terraform/providers
2024-05-07T09:14:25.431+0100 [TRACE] getproviders.SearchLocalDirectory: found registry.opentofu.org/cloudflare/cloudflare v4.30.0 for darwin_arm64 at .terraform/providers/registry.opentofu.org/cloudflare/cloudflare/4.30.0/darwin_arm64
2024-05-07T09:14:25.431+0100 [TRACE] getproviders.SearchLocalDirectory: found registry.opentofu.org/dopplerhq/doppler v1.7.0 for darwin_arm64 at .terraform/providers/registry.opentofu.org/dopplerhq/doppler/1.7.0/darwin_arm64
2024-05-07T09:14:25.432+0100 [TRACE] getproviders.SearchLocalDirectory: found registry.opentofu.org/hashicorp/archive v2.4.2 for darwin_arm64 at .terraform/providers/registry.opentofu.org/hashicorp/archive/2.4.2/darwin_arm64
2024-05-07T09:14:25.432+0100 [TRACE] getproviders.SearchLocalDirectory: found registry.opentofu.org/hashicorp/aws v5.46.0 for darwin_arm64 at .terraform/providers/registry.opentofu.org/hashicorp/aws/5.46.0/darwin_arm64
2024-05-07T09:14:25.432+0100 [TRACE] getproviders.SearchLocalDirectory: found registry.opentofu.org/hashicorp/google v5.25.0 for darwin_arm64 at .terraform/providers/registry.opentofu.org/hashicorp/google/5.25.0/darwin_arm64
2024-05-07T09:14:25.432+0100 [TRACE] getproviders.SearchLocalDirectory: found registry.opentofu.org/hashicorp/random v3.6.1 for darwin_arm64 at .terraform/providers/registry.opentofu.org/hashicorp/random/3.6.1/darwin_arm64
2024-05-07T09:14:25.432+0100 [TRACE] getproviders.SearchLocalDirectory: found registry.opentofu.org/vercel/vercel v1.6.0 for darwin_arm64 at .terraform/providers/registry.opentofu.org/vercel/vercel/1.6.0/darwin_arm64
2024-05-07T09:14:25.433+0100 [TRACE] getproviders.SearchLocalDirectory: found registry.terraform.io/cloudflare/cloudflare v4.30.0 for darwin_arm64 at .terraform/providers/registry.terraform.io/cloudflare/cloudflare/4.30.0/darwin_arm64
2024-05-07T09:14:25.433+0100 [TRACE] getproviders.SearchLocalDirectory: found registry.terraform.io/dopplerhq/doppler v1.7.0 for darwin_arm64 at .terraform/providers/registry.terraform.io/dopplerhq/doppler/1.7.0/darwin_arm64
2024-05-07T09:14:25.433+0100 [TRACE] getproviders.SearchLocalDirectory: found registry.terraform.io/hashicorp/archive v2.4.2 for darwin_arm64 at .terraform/providers/registry.terraform.io/hashicorp/archive/2.4.2/darwin_arm64
2024-05-07T09:14:25.433+0100 [TRACE] getproviders.SearchLocalDirectory: found registry.terraform.io/hashicorp/aws v5.46.0 for darwin_arm64 at .terraform/providers/registry.terraform.io/hashicorp/aws/5.46.0/darwin_arm64
2024-05-07T09:14:25.433+0100 [TRACE] getproviders.SearchLocalDirectory: found registry.terraform.io/hashicorp/google v5.25.0 for darwin_arm64 at .terraform/providers/registry.terraform.io/hashicorp/google/5.25.0/darwin_arm64
2024-05-07T09:14:25.433+0100 [TRACE] getproviders.SearchLocalDirectory: found registry.terraform.io/hashicorp/random v3.6.1 for darwin_arm64 at .terraform/providers/registry.terraform.io/hashicorp/random/3.6.1/darwin_arm64
2024-05-07T09:14:25.433+0100 [TRACE] getproviders.SearchLocalDirectory: found registry.terraform.io/vercel/vercel v1.6.0 for darwin_arm64 at .terraform/providers/registry.terraform.io/vercel/vercel/1.6.0/darwin_arm64
2024-05-07T09:14:25.433+0100 [TRACE] providercache.fillMetaCache: including .terraform/providers/registry.opentofu.org/cloudflare/cloudflare/4.30.0/darwin_arm64 as a candidate package for registry.opentofu.org/cloudflare/cloudflare 4.30.0
2024-05-07T09:14:25.433+0100 [TRACE] providercache.fillMetaCache: including .terraform/providers/registry.opentofu.org/hashicorp/archive/2.4.2/darwin_arm64 as a candidate package for registry.opentofu.org/hashicorp/archive 2.4.2
2024-05-07T09:14:25.433+0100 [TRACE] providercache.fillMetaCache: including .terraform/providers/registry.opentofu.org/hashicorp/random/3.6.1/darwin_arm64 as a candidate package for registry.opentofu.org/hashicorp/random 3.6.1
2024-05-07T09:14:25.433+0100 [TRACE] providercache.fillMetaCache: including .terraform/providers/registry.terraform.io/hashicorp/random/3.6.1/darwin_arm64 as a candidate package for registry.terraform.io/hashicorp/random 3.6.1
2024-05-07T09:14:25.433+0100 [TRACE] providercache.fillMetaCache: including .terraform/providers/registry.opentofu.org/dopplerhq/doppler/1.7.0/darwin_arm64 as a candidate package for registry.opentofu.org/dopplerhq/doppler 1.7.0
2024-05-07T09:14:25.434+0100 [TRACE] providercache.fillMetaCache: including .terraform/providers/registry.terraform.io/hashicorp/google/5.25.0/darwin_arm64 as a candidate package for registry.terraform.io/hashicorp/google 5.25.0
2024-05-07T09:14:25.434+0100 [TRACE] providercache.fillMetaCache: including .terraform/providers/registry.opentofu.org/vercel/vercel/1.6.0/darwin_arm64 as a candidate package for registry.opentofu.org/vercel/vercel 1.6.0
2024-05-07T09:14:25.434+0100 [TRACE] providercache.fillMetaCache: including .terraform/providers/registry.terraform.io/cloudflare/cloudflare/4.30.0/darwin_arm64 as a candidate package for registry.terraform.io/cloudflare/cloudflare 4.30.0
2024-05-07T09:14:25.434+0100 [TRACE] providercache.fillMetaCache: including .terraform/providers/registry.terraform.io/dopplerhq/doppler/1.7.0/darwin_arm64 as a candidate package for registry.terraform.io/dopplerhq/doppler 1.7.0
2024-05-07T09:14:25.434+0100 [TRACE] providercache.fillMetaCache: including .terraform/providers/registry.terraform.io/hashicorp/aws/5.46.0/darwin_arm64 as a candidate package for registry.terraform.io/hashicorp/aws 5.46.0
2024-05-07T09:14:25.434+0100 [TRACE] providercache.fillMetaCache: including .terraform/providers/registry.terraform.io/vercel/vercel/1.6.0/darwin_arm64 as a candidate package for registry.terraform.io/vercel/vercel 1.6.0
2024-05-07T09:14:25.434+0100 [TRACE] providercache.fillMetaCache: including .terraform/providers/registry.opentofu.org/hashicorp/aws/5.46.0/darwin_arm64 as a candidate package for registry.opentofu.org/hashicorp/aws 5.46.0
2024-05-07T09:14:25.434+0100 [TRACE] providercache.fillMetaCache: including .terraform/providers/registry.opentofu.org/hashicorp/google/5.25.0/darwin_arm64 as a candidate package for registry.opentofu.org/hashicorp/google 5.25.0
2024-05-07T09:14:25.434+0100 [TRACE] providercache.fillMetaCache: including .terraform/providers/registry.terraform.io/hashicorp/archive/2.4.2/darwin_arm64 as a candidate package for registry.terraform.io/hashicorp/archive 2.4.2
2024-05-07T09:14:25.445+0100 [TRACE] providercache.fillMetaCache: using cached result from previous scan of .terraform/providers
2024-05-07T09:14:25.827+0100 [TRACE] providercache.fillMetaCache: using cached result from previous scan of .terraform/providers
2024-05-07T09:14:25.912+0100 [TRACE] providercache.fillMetaCache: using cached result from previous scan of .terraform/providers
2024-05-07T09:14:25.923+0100 [TRACE] providercache.fillMetaCache: using cached result from previous scan of .terraform/providers
2024-05-07T09:14:25.934+0100 [TRACE] providercache.fillMetaCache: using cached result from previous scan of .terraform/providers
2024-05-07T09:14:25.962+0100 [TRACE] providercache.fillMetaCache: using cached result from previous scan of .terraform/providers
2024-05-07T09:14:25.973+0100 [DEBUG] checking for provisioner in "."
2024-05-07T09:14:25.976+0100 [DEBUG] checking for provisioner in "/opt/homebrew/bin"
2024-05-07T09:14:25.976+0100 [TRACE] Meta.Backend: backend *gcs.Backend does not support operations, so wrapping it in a local backend

!!!!!!!!!!!!!!!!!!!!!!!!!!! OPENTOFU CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

OpenTofu crashed! This is always indicative of a bug within OpenTofu.
Please report the crash with OpenTofu[1] so that we can fix this.

When reporting bugs, please include your OpenTofu version, the stack trace
shown below, and any additional information which may help replicate the issue.

[1]: https://github.com/opentofu/opentofu/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! OPENTOFU CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

runtime error: invalid memory address or nil pointer dereference
goroutine 1 [running]:
runtime/debug.Stack()
	runtime/debug/stack.go:24 +0x64
runtime/debug.PrintStack()
	runtime/debug/stack.go:16 +0x1c
github.com/opentofu/opentofu/internal/logging.panicHandler({0x1078b8020, 0x1095f9e70}, {0x0, 0x0, 0x0})
	github.com/opentofu/opentofu/internal/logging/panic.go:96 +0xcc
github.com/opentofu/opentofu/internal/logging.PanicHandler()
	github.com/opentofu/opentofu/internal/logging/panic.go:51 +0xd4
panic({0x1078b8020?, 0x1095f9e70?})
	runtime/panic.go:770 +0x124
github.com/opentofu/opentofu/internal/states/statefile.Read({0x107d81460?, 0x14000944cc0?}, {0x0, 0x0})
	github.com/opentofu/opentofu/internal/states/statefile/read.go:79 +0x150
github.com/opentofu/opentofu/internal/states/remote.(*State).refreshState(0x14000888070)
	github.com/opentofu/opentofu/internal/states/remote/state.go:167 +0x104
github.com/opentofu/opentofu/internal/states/remote.(*State).RefreshState(0x14000232100?)
	github.com/opentofu/opentofu/internal/states/remote/state.go:147 +0xa8
github.com/opentofu/opentofu/internal/backend/remote-state/gcs.(*Backend).StateMgr(0x14000232100, {0x106cf5d9d?, 0x107d81300?})
	github.com/opentofu/opentofu/internal/backend/remote-state/gcs/backend_state.go:104 +0x9c
github.com/opentofu/opentofu/internal/backend/local.(*Local).StateMgr(0x109600de0?, {0x106cf5d9d?, 0x0?})
	github.com/opentofu/opentofu/internal/backend/local/backend.go:252 +0x310
github.com/opentofu/opentofu/internal/command.(*UnlockCommand).Run(0x1400086ac40, {0x1400019c2f0?, 0x0?, 0x0?})
	github.com/opentofu/opentofu/internal/command/unlock.go:82 +0x5f8
github.com/mitchellh/cli.(*CLI).Run(0x140008583c0)
	github.com/mitchellh/cli@v1.1.5/cli.go:262 +0x4a8
main.realMain()
	github.com/opentofu/opentofu/cmd/tofu/main.go:347 +0x1a1c
main.main()
	github.com/opentofu/opentofu/cmd/tofu/main.go:66 +0x1c
```

### Expected Behavior

Remove state lock

### Actual Behavior

Crashes

### Steps to Reproduce

1. Cause a state lock
2. `tofu force-unlock 1714841031528627`

**Comment by cam72cam** [comment]
Looks like I made a bad assumption in https://github.com/opentofu/opentofu/blob/5ab6167/internal/command/unlock.go#L67.  Will get this fixed and out in 1.7.1.

**Comment by cam72cam** [comment]
This does not impact all backends.  The S3 backend does not require encryption to function, but it appears that the gcs backend does.  I will also be checking a few other locations in the command package for gcs crashes as well.
